### PR TITLE
Move page.js to node

### PR DESCRIPTION
**Pull request by grafixeyehero**
**Note: need test for tv and mobile platform**

And this still need some clean up

### CODE REVIEW

File: src/components/dialogHelper/dialogHelper.js [hunk 1]
```diff
@@ -172,7 +172,7 @@ define(['appRouter', 'focusManager', 'browser', 'layoutManager', 'inputManager',
     }
 
     function isHistoryEnabled(dlg) {
-        return dlg.getAttribute('data-history') === 'true';
+        return dlg.getAttribute('data-history') === 'false';
```
Comment (JustAMan) [inline src/components/dialogHelper/dialogHelper.js]: wtf? :)

File: src/components/appRouter.js [hunk 2]
```diff
@@ -666,7 +669,8 @@ define(['loading', 'globalize', 'events', 'viewManager', 'layoutManager', 'skinM
 
     function pushState(state, title, url) {
         state.navigate = false;
-        page.pushState(state, title, url);
+        history.pushState(state, title, url);
```
Comment (JustAMan) [inline src/components/appRouter.js]: where is `history` defined?

Comment (grafixeyehero) [reply src/components/appRouter.js]: https://developer.mozilla.org/en-US/docs/Web/API/History/pushState

File: src/components/dialogHelper/dialogHelper.js (hunk 1 above)
Comment (grafixeyehero) [reply src/components/dialogHelper/dialogHelper.js]: What's wrong with this?

Comment (dmitrylyzo) [reply src/components/dialogHelper/dialogHelper.js]: > What's wrong with this?

logic reverse

Comment (grafixeyehero) [reply src/components/dialogHelper/dialogHelper.js]: https://github.com/jellyfin/jellyfin-web/pull/871/commits/4fca1c15638e9f0cd4fc8317fe7d8baef115da66

File: src/components/appRouter.js [hunk 3]
```diff
@@ -562,7 +562,10 @@ define(['loading', 'globalize', 'events', 'viewManager', 'layoutManager', 'skinM
         if (!document.querySelector('.dialogContainer') && startPages.indexOf(curr.type) !== -1) {
             return false;
         }
-        return page.canGoBack();
+        if (enableHistory) {
```
Comment (dmitrylyzo) [inline src/components/appRouter.js]:
```suggestion
        if (enableHistory()) {
```

Comment (grafixeyehero) [reply src/components/appRouter.js]: https://github.com/jellyfin/jellyfin-web/pull/871/commits/08afa4c0ca07487ee3763201726af6472335ded1

**Comment by dmitrylyzo** [comment]
`enableHistory` and `enableNativeHistory` are still questionable.
I looked how it was before. Result of `enableHistory` function is passed to `page` constructor, then to `page.start`. Then it becomes the result of `page.enableNativeHistory`. So, `enableHistory` = `enableNativeHistory`. :confused:

**Comment by MrTimscampi** [comment]
Do note that page.js doesn't seem to have any parameter relating to history anymore, so I think it was dropped (and can thus be entirely removed).

All our clients support the HTML5 history API anyway.

In fact, the official documentation explicitly states that the history API should be polyfilled if the clients don't support it.

**Comment by JustAMan** [comment]
> `enableHistory` and `enableNativeHistory` are still questionable.
> I looked how it was before. Result of `enableHistory` function is passed to `page` constructor, then to `page.start`. Then it becomes the result of `page.enableNativeHistory`. So, `enableNativeHistory` = `enableHistory`. 😕



> Do note that page.js doesn't seem to have any parameter relating to history anymore, so I think it was dropped (and can thus be entirely removed).
> 
> All our clients support the HTML5 history API anyway.

it might have something to do with TVs being the clients... @dmitrylyzo your say?

**Comment by dmitrylyzo** [comment]
> it might have something to do with TVs being the clients...

`enableHistory` (and therefore `enableNativeHistory` - before PR) returned `true` for almost every device. So old `page.js` used `history` almost always. New `page.js` uses `history` if one exists. It seems that Xbox One has forcibly disabled history.
https://github.com/jellyfin/jellyfin-web/blob/9162f53b98872c60af923bccd06f761d48872639/src/components/appRouter.js#L359-L362

Above, I mean that now `enableHistory` and `enableNativeHistory` are not in sync.

Here is the only place where it is used:
https://github.com/jellyfin/jellyfin-web/blob/9162f53b98872c60af923bccd06f761d48872639/src/components/dialogHelper/dialogHelper.js#L378-L380
And now it is turned off, so dialogs do not use history, but navigation uses.

**Pull request merged**